src/main/java/io/strimzi/ValidatingWebhook.java

```diff
@@ -74,12 +78,10 @@ public AdmissionReview webhook(AdmissionReview review) {
         AdmissionRequest request = review.getRequest();
         ObjectMeta evictionMetadata = extractEvictionMetadata(request);
 
-        if (evictionMetadata != null)   {
-            if (drainZooKeeper && ZOOKEEPER_PATTERN.matcher(evictionMetadata.getName()).matches()
-                    || drainKafka && KAFKA_PATTERN.matcher(evictionMetadata.getName()).matches()) {
+        if (evictionMetadata != null) {
+            if ((drainKafka || drainZooKeeper) && (matchingLabel(evictionMetadata.getLabels()))) {
```

Comment:
I think this does not distinguish between Kafka and ZooKeeper being enabled. For example, when you enable it only for Kafka (i.e. `drainZooKeeper` is `false`), it would still pass the first part of the condition (`drainKafka || drainZooKeeper`) and handle the ZooKeeper pod. So this seems wrong.